<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from yaskawa_complete.xacro         | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="wgs32_camera">
  <!--degrees to radians-->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/base.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/base.dae"/>
      </geometry>
    </collision>
    <inertial>
      <!-- <origin rpy="0 0 0" xyz="-0.025918 -0.000376 0.076105"/> -->
      <mass value="9.23913043478"/>
      <inertia ixx="0.0378685009058" ixy="0" ixz="0" iyy="0.0465867830616" iyz="0" izz="0.0514065294384"/>
    </inertial>
  </link>
  <link name="link_s">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_s.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_s.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.36956521739"/>
      <inertia ixx="0.1285284" ixy="0" ixz="0" iyy="0.127697251956" iyz="0" izz="0.0186623672464"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0046451930 -0.018976267"/>
    </inertial>
  </link>
  <joint name="joint_s" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.31"/>
    <parent link="base_link"/>
    <child link="link_s"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-3.14159274" upper="3.14159274" velocity="3.4906586"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="link_l">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_l.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_l.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="7.60869565217"/>
      <inertia ixx="0.12248884058" ixy="0" ixz="0" iyy="0.117554616667" iyz="0" izz="0.0177495804348"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.11422130 0.0 0.018935113"/>
    </inertial>
  </link>
  <joint name="joint_l" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_s"/>
    <child link="link_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" lower="-1.9198622300000001" upper="1.9198622300000001" velocity="3.4906586"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="link_e">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_e.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_e.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.30434782609"/>
      <inertia ixx="0.0122899381739" ixy="0" ixz="0" iyy="0.0117127787826" iyz="0" izz="0.00344401234783"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.011901001 -0.036023969"/>
    </inertial>
  </link>
  <joint name="joint_e" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.27"/>
    <parent link="link_l"/>
    <child link="link_e"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="-2.9670598100000003" upper="2.9670598100000003" velocity="3.4906586"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="link_u">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_u.dae"/>
      </geometry>
    </visual>
    <collision>
      axis
      
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_u.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.08695652174"/>
      <inertia ixx="0.0110207115768" ixy="0" ixz="0" iyy="0.00996803594202" iyz="0" izz="0.00214043853333"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.015659426 0.13525753"/>
    </inertial>
  </link>
  <joint name="joint_u" type="revolute">
    <origin rpy="0 0 0" xyz="0.085 0 0"/>
    <parent link="link_e"/>
    <child link="link_u"/>
    <axis xyz="0 -1 0"/>
    <limit effort="0" lower="-1.57079637" upper="2.007128695" velocity="3.4906586"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="link_r">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_r.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_r.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.978260869565"/>
      <inertia ixx="0.0108400832201" ixy="0" ixz="0" iyy="0.010708873981" iyz="0" izz="0.00140456861413"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.011271807 -0.0416"/>
    </inertial>
  </link>
  <joint name="joint_r" type="revolute">
    <origin rpy="0 0 0" xyz="0.27 0 0.06"/>
    <parent link="link_u"/>
    <child link="link_r"/>
    <axis xyz="-1 0 0"/>
    <limit effort="0" lower="-1.57079637" upper="1.57079637" velocity="3.4906586"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="link_b">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_b.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_b.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.869565217391"/>
      <inertia ixx="0.00814341869565" ixy="0" ixz="0" iyy="0.00709473869565" iyz="0" izz="0.000736577391304"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0074364051 0.078721289"/>
    </inertial>
  </link>
  <joint name="joint_b" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_r"/>
    <child link="link_b"/>
    <axis xyz="0 -1 0"/>
    <limit effort="0" lower="-1.9198622300000001" upper="1.9198622300000001" velocity="4.01425739"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="link_t">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/visual/link_t.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_yaskawa_ros/motoman_sia5d_support/meshes/sia5d/collision/link_t.dae"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.54347826087"/>
      <inertia ixx="0.003" ixy="0" ixz="0" iyy="0.003" iyz="0" izz="0.001"/>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    </inertial>
  </link>
  <joint name="joint_t" type="revolute">
    <origin rpy="0 0 0" xyz="0.134 -0.0078 0"/>
    <parent link="link_b"/>
    <child link="link_t"/>
    <axis xyz="-1 0 0"/>
    <limit effort="0" lower="-1.57079637" upper="1.57079637" velocity="6.10865255"/>
    <dynamics damping="0.7" friction="0.7"/>
  </joint>
  <link name="tool0"/>
  <joint name="link_t-tool0" type="fixed">
    <origin rpy="-1.57079637 -1.57079637 -1.57079637" xyz="0.011 0 0"/>
    <parent link="link_t"/>
    <child link="tool0"/>
  </joint>
  <!-- transmission -->
  <!-- transmission -->
  <transmission name="transmission_s">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_s">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_s">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>300</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_l">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_l">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_l">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>300</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_e">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_e">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_e">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_u">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_u">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_u">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>50</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_r">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_r">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_r">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>30</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_b">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_b">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_b">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>30</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="transmission_t">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_t">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_t">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>10</mechanicalReduction>
    </actuator>
  </transmission>
  <!--degrees to radians-->
  <link name="base_link_gripper">
    <inertial>
      <origin rpy="0 0 0" xyz="0.03129440494284765 -0.0236275456406487 0.0633242924961397"/>
      <mass value="0.49962194311335323"/>
      <inertia ixx="0.001045" ixy="6.1e-05" ixz="0.000195" iyy="0.001292" iyz="-0.000254" izz="0.000438"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white1"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="camera_ext">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.03 0.032 0.11"/>
      <mass value="0.49962194311335323"/>
      <inertia ixx="0.001045" ixy="6.1e-05" ixz="0.000195" iyy="0.001292" iyz="-0.000254" izz="0.000438"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/extension_camera_support.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white1"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/extension_camera_support.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="ext_finger_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.005297128503583257 -0.015412565841978962 -0.010725009096658585"/>
      <mass value="0.17786635698192355"/>
      <inertia ixx="4.9e-05" ixy="8e-06" ixz="-0.0" iyy="3.1e-05" iyz="-0.0" izz="3.9e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/finger_ext.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white1"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/finger_ext.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="ext_finger_2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.005550196974422999 -0.015531770123516614 -0.010351884823019339"/>
      <mass value="0.17786635698192355"/>
      <inertia ixx="4.9e-05" ixy="-8e-06" ixz="-0.0" iyy="3.1e-05" iyz="0.0" izz="3.9e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/finger_ext.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="white1"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/gripper_description_ros2/meshes/finger_ext.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="camera_ext_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.0295 0.0327 0.1132"/>
    <parent link="base_link_gripper"/>
    <child link="camera_ext"/>
  </joint>
  <joint name="jaw_ext_1" type="fixed">
    <origin rpy="0 0 -0.01" xyz="0 0.005 0"/>
    <parent link="Base_Jaw_1"/>
    <child link="ext_finger_1"/>
  </joint>
  <joint name="jaw_ext_2" type="fixed">
    <origin rpy="0 0 -0.01" xyz="0 0.005 0"/>
    <parent link="Base_Jaw_2"/>
    <child link="ext_finger_2"/>
  </joint>
  <link name="WSG-32_Body">
    <inertial>
      <origin rpy="0 0 0" xyz="0.03574340512333041 -0.027873336366329354 -0.012493199179186804"/>
      <mass value="1.506317966969221"/>
      <inertia ixx="0.000527" ixy="5.2e-05" ixz="0.0" iyy="0.001724" iyz="-0.0" izz="0.002034"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/wsg32_description_ros2/meshes/WSG-32_Body.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/wsg32_description_ros2/meshes/WSG-32_Body.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Base_Jaw_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.08503018494095639 -0.06013843252995674 -0.0033036600416418175"/>
      <mass value="0.048005219849158315"/>
      <inertia ixx="1.1e-05" ixy="5e-06" ixz="3e-06" iyy="1.1e-05" iyz="-3e-06" izz="1.3e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/wsg32_description_ros2/meshes/Base_Jaw.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/wsg32_description_ros2/meshes/Base_Jaw.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Base_Jaw_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.025579123197365217 -0.06013843252995674 0.004705428018545392"/>
      <mass value="0.048005219849158315"/>
      <inertia ixx="1.1e-05" ixy="-5e-06" ixz="3e-06" iyy="1.1e-05" iyz="3e-06" izz="1.3e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/wsg32_description_ros2/meshes/Base_Jaw.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/wsg32_description_ros2/meshes/Base_Jaw.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_to_wsg" type="fixed">
    <origin rpy="3.14 0 -1.57" xyz="-0.0002 -0.0076 -0.0007"/>
    <parent link="base_link_gripper"/>
    <child link="WSG-32_Body"/>
  </joint>
  <joint name="right_joint" type="prismatic">
    <origin rpy="0 -1.570 3.140" xyz="0.062 -0.039 0.001"/>
    <parent link="WSG-32_Body"/>
    <child link="Base_Jaw_1"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="100.0" lower="-0.034" upper="0.0" velocity="100.0"/>
    <dynamics damping="1.0"/>
  </joint>
  <joint name="left_joint" type="prismatic">
    <origin rpy="0 1.570 0" xyz="0.062 0.038 0.001"/>
    <parent link="WSG-32_Body"/>
    <child link="Base_Jaw_2"/>
    <axis xyz="0.0 -1.0 -0.0"/>
    <limit effort="100.0" lower="-0.034" upper="0.0" velocity="100.0"/>
    <mimic joint="right_joint" multiplier="-1" offset="0"/>
    <dynamics damping="1.0"/>
  </joint>
  <material name="silver">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1.0"/>
  </material>
  <material name="white1">
    <color rgba="0.95 0.937 0.937 1.0"/>
  </material>
  <joint name="$tactile_1_joint_ext_finger_1" type="fixed">
    <parent link="ext_finger_1"/>
    <child link="tactile_1"/>
    <origin rpy="-1.5902 0.01 -1.5511" xyz="-0.0379 0.02 -0.0987"/>
  </joint>
  <joint name="$tactile_2_joint_ext_finger_2" type="fixed">
    <parent link="ext_finger_2"/>
    <child link="tactile_2"/>
    <origin rpy="-1.5805 0.01 -1.55" xyz="-0.0369 0.02 -0.099"/>
  </joint>
  <link name="tactile_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0028157142675214186 -0.04814932465420843 -0.010573307928346813"/>
      <mass value="0.4046663410557083"/>
      <inertia ixx="0.000596" ixy="-3e-05" ixz="-0.0" iyy="5.6e-05" iyz="-1e-06" izz="0.000573"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_fingers_description_ros2/meshes/finger_refills.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_fingers_description_ros2/meshes/finger_refills.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="tactile_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.002830334344263878 -0.04812604253213226 -0.010675341666564392"/>
      <mass value="0.4046663410557083"/>
      <inertia ixx="0.000595" ixy="3.5e-05" ixz="-0.0" iyy="5.6e-05" iyz="-2e-06" izz="0.000573"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_fingers_description_ros2/meshes/finger_refills.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/workstation2/AlessioBenitoAlterani/Master-thesis/DenseFusion_ros/src/uclv_fingers_description_ros2/meshes/finger_refills.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1"/>
  </material>
  <!-- camera body, with origin at bottom screw mount -->
  <joint name="camera_joint" type="fixed">
    <origin rpy="3.14 0.12 -1.57" xyz="-0.0034 -0.0223 0.03"/>
    <parent link="camera_ext"/>
    <child link="camera_bottom_screw_frame"/>
  </joint>
  <link name="camera_bottom_screw_frame"/>
  <joint name="camera_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.010600000000000002 0.0175 0.0125"/>
    <parent link="camera_bottom_screw_frame"/>
    <child link="camera_link"/>
  </joint>
  <link name="camera_link">
    <visual>
      <!-- the mesh origin is at front plate in between the two infrared camera axes -->
      <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0.0043 -0.0175 0"/>
      <geometry>
        <mesh filename="package://realsense2_description/meshes/d435.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.0175 0"/>
      <geometry>
        <box size="0.02505 0.09 0.025"/>
      </geometry>
    </collision>
    <inertial>
      <!-- The following are not reliable values, and should not be used for modeling -->
      <mass value="0.072"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.003881243" ixy="0.0" ixz="0.0" iyy="0.000498940" iyz="0.0" izz="0.003879257"/>
    </inertial>
  </link>
  <!-- camera depth joints and links -->
  <joint name="camera_depth_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="camera_link"/>
    <child link="camera_depth_frame"/>
  </joint>
  <link name="camera_depth_frame"/>
  <joint name="camera_depth_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_depth_frame"/>
    <child link="camera_depth_optical_frame"/>
  </joint>
  <link name="camera_depth_optical_frame"/>
  <!-- camera left IR joints and links -->
  <joint name="camera_infra1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.0 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra1_frame"/>
  </joint>
  <link name="camera_infra1_frame"/>
  <joint name="camera_infra1_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_infra1_frame"/>
    <child link="camera_infra1_optical_frame"/>
  </joint>
  <link name="camera_infra1_optical_frame"/>
  <!-- camera right IR joints and links -->
  <joint name="camera_infra2_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.05 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra2_frame"/>
  </joint>
  <link name="camera_infra2_frame"/>
  <joint name="camera_infra2_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_infra2_frame"/>
    <child link="camera_infra2_optical_frame"/>
  </joint>
  <link name="camera_infra2_optical_frame"/>
  <!-- camera color joints and links -->
  <joint name="camera_color_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.015 0"/>
    <parent link="camera_link"/>
    <child link="camera_color_frame"/>
  </joint>
  <link name="camera_color_frame"/>
  <joint name="camera_color_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_color_frame"/>
    <child link="camera_color_optical_frame"/>
  </joint>
  <link name="camera_color_optical_frame"/>
  <link name="camera_accel_frame"/>
  <link name="camera_accel_optical_frame"/>
  <link name="camera_gyro_frame"/>
  <link name="camera_gyro_optical_frame"/>
  <joint name="camera_accel_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.01174 -0.00552 0.0051"/>
    <parent link="camera_link"/>
    <child link="camera_accel_frame"/>
  </joint>
  <joint name="camera_accel_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_accel_frame"/>
    <child link="camera_accel_optical_frame"/>
  </joint>
  <joint name="camera_gyro_joint" type="fixed">
    <origin rpy="0 0 0" xyz="-0.01174 -0.00552 0.0051"/>
    <parent link="camera_link"/>
    <child link="camera_gyro_frame"/>
  </joint>
  <joint name="camera_gyro_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_gyro_frame"/>
    <child link="camera_gyro_optical_frame"/>
  </joint>
  <!-- CAMERA REALSENSE D435i bottom-->
  <!-- <xacro:sensor_d435i parent="base_link_gripper" name="camera_bottom" use_nominal_extrinsics="true">
  <origin xyz="-0.033 0.023 0.094" rpy="0 0.378 -1.57"/>
</xacro:sensor_d435i> -->
  <!-- END EFFECTOR FINGERS -->
  <link name="ee_fingers"> </link>
  <joint name="tool0_ee" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.21"/>
    <parent link="tool0"/>
    <child link="ee_fingers"/>
  </joint>
  <joint name="tool0_joint_base_gripper" type="fixed">
    <origin rpy="-1.57079637 0 -1.57079637" xyz="0 0 0"/>
    <parent link="tool0"/>
    <child link="base_link_gripper"/>
  </joint>
</robot>

